js_modules/dagster-ui/packages/ui-core/src/assets/AssetHealthSummary.tsx

```diff
@@ -73,16 +109,150 @@ export const AssetHealthSummary = React.memo(
 );
 
 const Criteria = React.memo(
-  ({status, text}: {status: AssetHealthStatus | undefined; text: string}) => {
+  ({
+    status,
+    text,
+    metadata,
+    explanation,
+    assetKey,
+  }: {
+    assetKey: {path: string[]};
+    status: AssetHealthStatus | undefined;
+    text: string;
+    metadata?:
+      | AssetHealthCheckDegradedMetaFragment
+      | AssetHealthCheckWarningMetaFragment
+      | AssetHealthCheckUnknownMetaFragment
+      | AssetHealthMaterializationDegradedNotPartitionedMetaFragment
+      | AssetHealthMaterializationDegradedPartitionedMetaFragment
+      | AssetHealthMaterializationWarningPartitionedMetaFragment
+      | undefined
+      | null;
+    explanation?: string;
+  }) => {
     const {subStatusIconName, iconColor, textColor} = statusToIconAndColor[status ?? 'undefined'];
 
+    const derivedExplanation = useMemo(() => {
+      switch (metadata?.__typename) {
+        case 'AssetHealthCheckUnknownMeta':
+          if (metadata.numNotExecutedChecks > 0) {
+            return (
+              <Body>
+                {numberFormatter.format(metadata.numNotExecutedChecks)} /{' '}
+                {numberFormatter.format(metadata.totalNumChecks)}{' '}
+                <Link to={assetDetailsPathForKey(assetKey, {view: 'checks'})}>
+                  check
+                  {ifPlural(metadata.totalNumChecks, '', 's')}
+                </Link>{' '}
+                not executed
+              </Body>
+            );
+          }
+          return 'No checks executed';
+        case 'AssetHealthCheckDegradedMeta':
+          if (metadata.numWarningChecks > 0 && metadata.numFailedChecks > 0) {
+            return (
+              <Body>
+                {numberFormatter.format(metadata.numWarningChecks)}/
+                {numberFormatter.format(metadata.totalNumChecks)}{' '}
+                <Link to={assetDetailsPathForKey(assetKey, {view: 'checks'})}>
+                  check
+                  {ifPlural(metadata.totalNumChecks, '', 's')}
+                </Link>{' '}
+                warning, {numberFormatter.format(metadata.numFailedChecks)}/
+                {numberFormatter.format(metadata.totalNumChecks)}{' '}
+                <Link to={assetDetailsPathForKey(assetKey, {view: 'checks'})}>
+                  check
+                  {ifPlural(metadata.totalNumChecks, '', 's')}
+                </Link>{' '}
+                failed
+              </Body>
+            );
+          }
+          if (metadata.numWarningChecks > 0) {
+            return (
+              <Body>
+                {numberFormatter.format(metadata.numWarningChecks)}/
+                {numberFormatter.format(metadata.totalNumChecks)}{' '}
+                <Link to={assetDetailsPathForKey(assetKey, {view: 'checks'})}>
+                  check
+                  {ifPlural(metadata.totalNumChecks, '', 's')}
+                </Link>{' '}
```

Comment:
feels vaguely like a `checkStatus(metadata.numWarningChecks, metadata.totalNumChecks, 'warning')` style helper could make this more readable, hard to see what is changing between all these cases 

Probably not worth changing though